clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1608,6 +1789,37 @@ public OffsetAndMetadata committed(TopicPartition partition) {
         }
     }
 
+    /**
+     * Get metadata about partitions for all topics that the user is authorized to view. This method will issue a
+     * remote call to the server.
+     *
+     * @param timeout The maximum time this operation will block
+     *
+     * @return The map of topics and its partitions
+     * @throws org.apache.kafka.common.errors.TimeoutException if time spent blocking exceeds the {@code timeout}
+     * @throws org.apache.kafka.common.errors.WakeupException if {@link #wakeup()} is called before or while this
+     *             function is called
+     * @throws org.apache.kafka.common.errors.InterruptException if the calling thread is interrupted before or while
+     *             this function is called
+     * @throws org.apache.kafka.common.errors.TimeoutException if the topic metadata could not be fetched before
+     *             expiration of the configured request timeout
+     * @throws org.apache.kafka.common.KafkaException for any other unrecoverable errors
+     */
+    public Map<String, List<PartitionInfo>> listTopics(Duration timeout) {
+        acquireAndEnsureOpen();
+        try {
+            long currMillis = time.milliseconds();
+            Map<String, List<PartitionInfo>> topicMetadata = fetcher.getAllTopicMetadata(timeout.toMillis());
```

Comment:
`getAllTopicMetadata` already raises `TimeoutException`, so I don't think we need any of this logic.